cmd/deploy/deploy.go

```diff
@@ -197,13 +201,45 @@ func (dc *deployCommand) runDeploy(ctx context.Context, cwd string, opts *Option
 		log.Infof("could not create temporal kubeconfig %s", err)
 		return err
 	}
-
 	var err error
-	// Read manifest file with the commands to be executed
-	opts.Manifest, err = dc.getManifest(cwd, contextCMD.ManifestOptions{Name: opts.Name, Filename: opts.ManifestPath})
-	if err != nil {
-		log.Infof("could not find manifest file to be executed: %s", err)
-		return err
+	if opts.Build && contextCMD.IsManifestV2Enabled() {
+		opts.Manifest, err = contextCMD.GetManifestV2(cwd, opts.ManifestPath)
+		if err != nil {
+			return err
+		}
+		log.Debug("found okteto manifest")
+
+		if opts.Manifest.Deploy == nil {
+			return fmt.Errorf("found okteto manifest, but no deploy commands where defined")
+		}
```

Comment:
Maybe this check can be moved after the if/else (line 244). I guess line 238 can also return a manifest without Deploy section

Reply:
I added the check because when i tested with empty deploy/not deploy declaration, a panic was triggered in line 267. There are two reasons behind this:
- one that, in case there is no deploy declaration, the command okteto deploy --build makes little sense, so why would i want to build the images if i dont even have a deploy command? so i'm returning at this point for that.
- other, I haven't tested all scenarios when "old" manifest is used, which will use the dc.gestManifest, this always returns a deploy section with commands depending of the "manifest" found (stack, pipeline, helm, etc...) so its kindof... never empty? i want to keep the changes as small as possible.

would it be ok to add the check (will be at the two lines) on line 266 to catch this possible panic? @ifbyol

Reply:
I would add it, but before starting the proxy, line 244 should be fine, or even inside the else branch (line 242)